src/modules/adaptors/Sommelier/ERC4626Adaptor.sol

```diff
@@ -0,0 +1,175 @@
+     * @dev Important to verify that external receivers are allowed if receiver is not Cellar address.
+     * @param assets the amount of assets to withdraw from the ERC4626 position
+     * @param receiver address to send assets to'
+     * @param adaptorData data needed to withdraw from the ERC4626 position
+     * @param configurationData abi encoded bool indicating whether the position is liquid or not
+     */
+    function withdraw(
+        uint256 assets,
+        address receiver,
+        bytes memory adaptorData,
+        bytes memory configurationData
+    ) public virtual override {
+        // Check that position is setup to be liquid.
+        bool isLiquid = abi.decode(configurationData, (bool));
+        if (!isLiquid) revert BaseAdaptor__UserWithdrawsNotAllowed();
+
+        // Run external receiver check.
+        _externalReceiverCheck(receiver);
+
+        // Withdraw assets from `cellar`.
+        ERC4626 erc4626Vault = abi.decode(adaptorData, (ERC4626));
+        _verifyERC4626PositionIsUsed(address(erc4626Vault));
+        erc4626Vault.withdraw(assets, receiver, address(this));
+    }
+
+    /**
+     * @notice Cellar needs to call `maxWithdraw` to see if its assets are locked.
+     */
+    function withdrawableFrom(
+        bytes memory adaptorData,
+        bytes memory configurationData
+    ) public view virtual override returns (uint256) {
+        bool isLiquid = abi.decode(configurationData, (bool));
+        if (isLiquid) {
+            ERC4626 erc4626Vault = abi.decode(adaptorData, (ERC4626));
+            return erc4626Vault.maxWithdraw(msg.sender);
+        } else return 0;
+    }
+
+    /**
+     * @notice Uses ERC4626 `previewRedeem` to determine Cellars balance in ERC4626 position.
+     */
+    function balanceOf(bytes memory adaptorData) public view virtual override returns (uint256) {
+        ERC4626 erc4626Vault = abi.decode(adaptorData, (ERC4626));
+        return erc4626Vault.previewRedeem(erc4626Vault.balanceOf(msg.sender));
+    }
+
+    /**
+     * @notice Returns the asset the ERC4626 position uses.
+     */
+    function assetOf(bytes memory adaptorData) public view virtual override returns (ERC20) {
+        ERC4626 erc4626Vault = abi.decode(adaptorData, (ERC4626));
+        return ERC20(erc4626Vault.asset());
+    }
+
+    /**
+     * @notice This adaptor returns collateral, and not debt.
+     */
+    function isDebt() public pure virtual override returns (bool) {
+        return false;
+    }
+
+    //============================================ Strategist Functions ===========================================
+    /**
+     * @notice Allows strategists to deposit into ERC4626 positions.
+     * @dev Uses `_maxAvailable` helper function, see BaseAdaptor.sol
+     * @param erc4626Vault the ERC4626 to deposit `assets` into
+     * @param assets the amount of assets to deposit into `cellar`
+     */
+    function depositToCellar(ERC4626 erc4626Vault, uint256 assets) public {
```

Comment:
Can you change the strategist withdraw and deposit functions to be `depositToVault` and `withdrawFromVault`

Reply:
Done!